sn_networking/src/version.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use lazy_static::lazy_static;
+
+lazy_static! {
+    /// The node version used during Identify Behaviour.
+    pub static ref IDENTIFY_NODE_VERSION_STR: String =
+        format!(
+            "safe/node/{}/{}",
+            get_testnet_channel(),
+            get_truncate_version_str()
+        );
+
+    /// The client version used during Identify Behaviour.
+    pub static ref IDENTIFY_CLIENT_VERSION_STR: String =
+        format!(
+            "safe/client/{}/{}",
+            get_testnet_channel(),
+            get_truncate_version_str()
+        );
+
+    /// / first version for the req/response protocol
+    pub static ref REQ_RESPONSE_VERSION_STR: String =
+        format!(
+            "/safe/{}/{}",
+            get_testnet_channel(),
+            get_truncate_version_str()
+        );
+
+
+    /// The identify protocol version
+    pub static ref IDENTIFY_PROTOCOL_STR: String =
+        format!(
+            "safe/{}/{}",
+            get_testnet_channel(),
+            get_truncate_version_str()
+        );
+
+
+}
+
+// Protocol support shall be downward compatible for patch only version update.
+// i.e. versions of `A.B.X` shall be considered as a same protocol of `A.B`
+fn get_truncate_version_str() -> &'static str {
+    let version_str = env!("CARGO_PKG_VERSION");
+    if version_str.matches('.').count() == 2 {
+        match version_str.rfind('.') {
+            Some(pos) => &version_str[..pos],
+            None => version_str,
+        }
+    } else {
+        version_str
+    }
+}
+
+// Return the testnet channel name from the git branch.
+// Stable branch is considered as the "beta" network.
+fn get_testnet_channel() -> &'static str {
+    match sn_build_info::git_branch() {
+        "stable" => "beta",
+        _ => "alpha",
```

Comment:
Alternative approach.

If the branch is `stable` or `beta`, do not apply anything.

Otherwise, apply the full branch name? (Would this mean local dev on a branch will be incompatible with a testnet? If so, we probably don't want this on _always_, and could only do so if there's an ENV_VAR: `RESTRICTED_NETWORK` or something?)